api/include/opentelemetry/context/context.h

```diff
@@ -133,8 +130,17 @@ class Context
       key_        = new char[key.size()];
       key_length_ = key.size();
       memcpy(key_, key.data(), key.size() * sizeof(char));
-      value_ = value;
       next_  = nostd::shared_ptr<DataList>{nullptr};
+      value_ = value;
+    }
+
+    DataList(const DataList &other)
+        : key_(new char[other.key_length_]),
+          next_(other.next_),
+          key_length_(other.key_length_),
+          value_(other.value_)
+    {
+      memcpy(key_, other.key_, other.key_length_ * sizeof(char));
```

Comment:
```suggestion
      std::memcpy(key_, other.key_, other.key_length_ * sizeof(char));
```

because we `#include <cstring>`, not `<string.h>`